src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -77,6 +79,58 @@ internal static ParseResult ParseScript(string script, ILanguageSettings loc = n
             return new ParseResult(parsetree, errors, errors?.Any() ?? false, parser._comments, parser._before, parser._after);
         }
 
+        public static ParseFormulasResult ParseFormulasScript(string script, ILanguageSettings loc = null, Flags flags = Flags.None)
+        {
+            Contracts.AssertValue(script);
+            Contracts.AssertValueOrNull(loc);
+
+            Token[] formulaTokens = TokenizeScript(script, loc, flags);
+            TexlParser parser = new TexlParser(formulaTokens, Flags.NamedFormulas);
+
+            return parser.ParseFormulas();
+        }
+
+        private ParseFormulasResult ParseFormulas()
+        {
+            Dictionary<DName, TexlNode> namedFormulas = new Dictionary<DName, TexlNode>();
+            while (_curs.TokCur.Kind != TokKind.Eof)
+            {
+                // Verify identifier
+                Token thisIdentifier = TokEat(TokKind.Ident);
+                if (thisIdentifier != null)
+                {
+                    // Verify "="
+                    Token thisEq = TokEat(TokKind.Equ);
+                    if (thisEq != null)
+                    {
+                        // Extract expression
+                        while (_curs.TidCur != TokKind.Semicolon)
+                        {
+                            // Check if we're at EOF before a semicolon is found
+                            if (_curs.TidCur == TokKind.Eof)
+                            {
+                                CreateError(_curs.TokCur, TexlStrings.ErrNamedFormula_MissingSemicolon);
+                                return new ParseFormulasResult(namedFormulas, _errors, _errors?.Any() ?? false);
```

Comment:
>, _errors?.Any() ?? false [](http://example.com/codeflow?start=85&length=25)

nit: 
this could be moved into the ctor, which simplifies the caller.   Every caller needs to pass this? (see line 131 too)

Reply:
Fixed